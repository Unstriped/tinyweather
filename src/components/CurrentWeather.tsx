import React from "react";
import { Weather } from "../types/weather-types";

interface WeatherProps {
  weather: Weather;
}

// Shows weather information and an matching image
const CurrentWeather: React.FC<WeatherProps> = ({ weather }) => {
  return (
    <>
      <div className="flex -ml-4">
        <img src={weather.condition.icon} alt={weather.condition.text} />
        <span className="text-4xl mt-2 mr-2 flex">
          {weather.temp_c}
          <span className="text-sm">Â°C</span>
        </span>
        <ul className="text-xs mt-2">
          <li>Wind Speed: {weather.wind_kph} km/h</li>
          <li>Humidity {weather.humidity}</li>
        </ul>
      </div>
      <p className="-mt-2 text-sm">{weather.condition.text}</p>
    </>
  );
};

export default CurrentWeather;
