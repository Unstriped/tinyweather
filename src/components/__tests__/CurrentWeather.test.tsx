import { render } from "@testing-library/react";
import React from "react";
import "@testing-library/jest-dom";
import CurrentWeather from "../CurrentWeather";
import { Weather } from "../../types/weather-types";

describe("Test if the LocationInput component is working as expected", () => {
  const weather = {
    condition: {
      text: "windy",
      icon: "//cdn.weatherapi.com/weather/64x64/night/116.png",
    },
    temp_c: 25.6,
    wind_kph: 16.9,
    humidity: 4,
  } as Weather;

  it("<CurrentWeather /> matches snapshot", () => {
    const component = render(<CurrentWeather weather={weather} />);
    expect(component.container).toMatchSnapshot();
  });

  it("<CurrentWeather /> renders without exploding", () => {
    const component = render(<CurrentWeather weather={weather} />);
    expect(component.getByText("Wind Speed: 16.9 km/h")).toBeInTheDocument();
  });
});
