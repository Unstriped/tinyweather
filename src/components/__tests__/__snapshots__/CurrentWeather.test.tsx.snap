// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the LocationInput component is working as expected <CurrentWeather /> matches snapshot 1`] = `
<div>
  <div
    class="flex -ml-4"
  >
    <img
      alt="windy"
      src="//cdn.weatherapi.com/weather/64x64/night/116.png"
    />
    <span
      class="text-4xl mt-2 mr-2 flex"
    >
      25.6
      <span
        class="text-sm"
      >
        Â°C
      </span>
    </span>
    <ul
      class="text-xs mt-2"
    >
      <li>
        Wind Speed: 
        16.9
         km/h
      </li>
      <li>
        Humidity 
        4
      </li>
    </ul>
  </div>
  <p
    class="-mt-2 text-sm"
  >
    windy
  </p>
</div>
`;
